src/api/use-user.api.ts

```diff
@@ -0,0 +1,32 @@
+import { supabase } from "../lib/supabase";
+import { UserModel } from "../models/user.model";
+import { showToast } from "../utils/toast";
+
+export const getUserByEmail = async (email: string) => {
```

Comment:
Might not be necessary for now, but [react-query](https://tanstack.com/query/latest) is great for this kind of thing, as it manages cache for you so you can use query results throughout the app